crates/red_knot_python_semantic/resources/mdtest/import/relative.md

```diff
@@ -121,13 +125,11 @@ X = 42
 ```
 
 ```py path=package/bar.py
-# TODO: support submodule imports
-from . import foo  # error: [unresolved-import]
+from . import foo
```

Comment:
I think this means that you can delete the last bullet point of the TODO here:

https://github.com/astral-sh/ruff/blob/7c2e7cf25e2b04f7b52052ed4c0710a593ebe4a1/crates/red_knot_python_semantic/src/types/infer.rs#L2244-L2252

Reply:
Oh, I think you can also get rid of the TODO comment here:

https://github.com/astral-sh/ruff/blob/dcb99cc8171aa579256f53ca3277a6ac524f5224/crates/red_knot_python_semantic/resources/mdtest/import/relative.md?plain=1#L133-L143

I don't think there's actually anything to do there? It looks to me like that test is actually testing that we emit a diagnostic if you attempt a relative import of a nonexistent submodule. So previously it passed for the wrong reasons, but I'm guessing with this PR it now passes for the right reasons...?

It would be great to add a short description to that test that clarifies what it's actually testing 😆

Reply:
> So previously it passed for the wrong reasons, but I'm guessing with this PR it now passes for the right reasons...?

Ha!  Yeah that looks right to me...  I was a bit surprised, and also a bit not surprised, to learn that that `from . import foo` syntax works to import a symbol from the current package.  Surprised in that it's a no-op, so why would you do that.  But not surprised in that it seems like the right behavior for that syntax.

Anyway, added a comment to hopefully clarify all of this.